### PR TITLE
Preparatory PR for setting unsandboxed process app IDs

### PR DESCRIPTION
**Pull request by mwleeds**
As discussed in #581, this is the work needed to make this changes possible.

### CODE REVIEW

File: src/xdp-utils.c [hunk 1]
```diff
@@ -929,6 +929,9 @@ xdp_get_alternate_document_path (const char *path, const char *app_id)
 {
   int len;
 
+  if (g_str_equal (app_id, ""))
```
Comment (GeorgesStavracas) [inline src/xdp-utils.c:932]: `g_str_equal()` is **not** NULL-safe, is it possible that `app_id` is NULL here? If so, it'd be more appropriate to use `app_id == NULL || g_str_equal(app_id, "")`, or perhaps `g_strcmp0()`

File: src/account.c [hunk 2]
```diff
@@ -95,7 +95,11 @@ send_response_in_thread_func (GTask        *task,
       g_autofree char *ruri = NULL;
       g_autoptr(GError) error = NULL;
 
-      ruri = register_document (image, xdp_app_info_get_id (request->app_info), FALSE, FALSE, FALSE, &error);
+      if (xdp_app_info_is_host (request->app_info))
+        ruri = g_strdup (image);
+      else
+        ruri = register_document (image, xdp_app_info_get_id (request->app_info), FALSE, FALSE, FALSE, &error);
+
```
Comment (GeorgesStavracas) [inline src/account.c:102]: I wonder if `register_document()` should receive `XdpAppInfo`, and implement this host/sandbox behaviour internally?

**Review by GeorgesStavracas — COMMENTED** [review]
Couple of comments

File: src/xdp-utils.c (hunk 1 above)
Comment (mwleeds) [reply src/xdp-utils.c:932]: No, `xdp_app_info_get_id` is guaranteed to return a non-NULL value. I have some docs written to document that guarantee that I'm going to push with the follow-up PR.

File: src/account.c (hunk 2 above)
Comment (mwleeds) [reply src/account.c:102]: The reason I didn't do that is there is no `XdpAppInfo` available in `open-uri.c`; we only have the app ID in that case.

**Pull request merged**